core/eolearn/core/core_tasks.py

```diff
@@ -272,3 +275,206 @@ def execute(self, eopatch):
+        """
+        raise NotImplementedError('map_method should be overridden.')
+
+
+class ZipFeaturesTask(EOTask):
+    """ Passes a set of input_features to a function, which returns a single features as a result and stores it in
+        the eopatch.
+
+        Example using inheritance:
+
+        .. code-block:: python
+
+            class CalculateFeatures(ZipFeaturesTask):
+                def map_function(self, *f):
+                    return f[0] / (f[1] + f[2])
+
+            calc = CalculateFeatures({FeatureType.DATA: ['f1', 'f2', 'f3']}, # input features
+                                     (FeatureType.MASK, 'm1'))               # output feature
+
+            result = calc(patch)
+
+        Example using lambda:
+
+        .. code-block:: python
+
+            calc = ZipFeaturesTask({FeatureType.DATA: ['f1', 'f2', 'f3']}, # input features
+                                   (FeatureType.MASK, 'm1'),               # output feature
+                                   lambda f0, f1, f2: f0 / (f1 + f2))      # function to apply to each feature
+
+            result = multiply(patch)
+    """
+    def __init__(self, input_features, output_feature, zip_function=None):
+        """
+        :param input_features: A collection of the input features to be mapped.
+        :type input_features: an object supported by eolearn.core.utilities.FeatureParser class
+        :param output_feature: An output feature object to which to assign the output data.
+        :type output_feature: an object supported by eolearn.core.utilities.FeatureParser class
+        :param zip_function: A function or lambda to be applied to the input data.
+        """
+        self.f_in = list(self._parse_features(input_features))
+        self.f_out = next(self._parse_features(output_feature)())
+        self.function = zip_function if zip_function else self.zip_method
+
+    def execute(self, eopatch):
+        """
+        :param eopatch: Source EOPatch from which to read the data of input features.
+        :type eopatch: EOPatch
+        :return: An eopatch with the additional zipped features.
+        :rtype: EOPatch
+        """
+        data = [eopatch[feature] for feature in self.f_in]
+
+        eopatch[self.f_out] = self.function(*data)
+
+        return eopatch
+
+    def zip_method(self, *f):
+        """A function that will be applied to the input features if overridden.
+
+        :raises NotImplementedError: When called and was neither overridden nor function argument was provided in
+        __init__.
+        """
+        raise NotImplementedError('zip_method should be overridden.')
+
+
+class MergeFeaturesTask(ZipFeaturesTask):
+    """ Merges multiple features together by concatenating their data along the last axis.
+    """
+    def zip_method(self, *f):
+        """Concatenates the data of features along the last axis.
+        """
+        return np.concatenate(f, axis=-1)
+
+
+class MoveBandsTask(EOTask):
```

Comment:
I am not sure about the name. The task doesn't really move data from one feature to another but rather extracts a part of the data and puts it to another feature. The initial feature stays the same.
Therefore maybe the name `ExtractBandsTask` might be better? Or something similar...

Reply:
Done.